### ISSUE TITLE
Flask Transactions that throw exceptions don't appear in Scout

### ISSUE THREAD

**Issue by itsderek23** [scoutapp/scout_apm_python#29]
If a Flask transaction throws an error, it doesn't appear that the transaction performance metrics appear in Scout. 

For example, in the following:

```python
@app.route('/boom_it')
def boom():
    random.hey()
```

`app.routes.boom` won't appear within Scout.

Note I had Sentry installed as well (those exceptions appear).

**Comment by itsderek23** [comment]
I'm seeing the same thing with Django views:

```py
def error(request):
  datetime.datetime.now()
  return HttpResponse("Bang!")

**Comment by dlanderson** [comment]
When we start a Flask request, [we start a 'Flask' layer](https://github.com/scoutapp/scout_apm_python/blob/master/src/scout_apm/flask/__init__.py#L76)

If the view function raises and exception, [we mark it as an error correctly and stop the 'Controller' span and then reraise](https://github.com/scoutapp/scout_apm_python/blob/master/src/scout_apm/flask/__init__.py#L116).

It seems that if the view function raises an exception, Flask will not get to the point of the ['process_response' in our code that stops the 'Flask' span.](https://github.com/scoutapp/scout_apm_python/blob/master/src/scout_apm/flask/__init__.py#L126).

We only stop a TrackedRequest and send it to the core agent when there are no spans left to stop. Because the 'Flask' span is never stopped, the TR never ends.

We can likely remove the 'Flask' span. This does expose a potential for a never ending TrackedRequest in other cases where a mismatch of Start vs Stop spans happens.

**Comment by dlanderson** [comment]
Resolved by https://github.com/scoutapp/scout_apm_python/pull/99
Will be released in v1.2.2
